pkg/rpc/server/server.go

```diff
@@ -1154,36 +1157,42 @@ func (s *Server) invokeContractVerify(reqParams request.Params) (interface{}, *r
 		return nil, responseErr
 	}
 
-	args := reqParams[1:2]
-	var tx *transaction.Transaction
+	bw := io.NewBufBinWriter()
+	if len(reqParams) > 1 {
+		args := reqParams[1] // second `invokecontractverify` argument is an array of arguments for `verify` method
+		if len(args.Value.([]request.Param)) > 0 {
+			err := request.EmitParams(bw, request.Params{args})
+			if err != nil {
+				return nil, response.NewRPCError("can't create witness invocation script", err.Error(), err)
+			}
+		}
+	}
+	invocationScript := bw.Bytes()
+
+	tx := &transaction.Transaction{Script: []byte{byte(opcode.RET)}} // need something in script
 	if len(reqParams) > 2 {
 		signers, witnesses, err := reqParams[2].GetSignersWithWitnesses()
 		if err != nil {
 			return nil, response.ErrInvalidParams
 		}
-		tx = &transaction.Transaction{
-			Signers: signers,
-			Scripts: witnesses,
-		}
+		tx.Signers = signers
+		tx.Scripts = witnesses
+	} else { // fill the only known signer - the contract with `verify` method
+		tx.Signers = []transaction.Signer{{Account: scriptHash}}
+		tx.Scripts = []transaction.Witness{{InvocationScript: invocationScript, VerificationScript: []byte{}}}
```

Comment:
Shouldn't at least `VerificationScript` be empty? https://github.com/nspcc-dev/neo-go/blob/master/pkg/core/blockchain.go#L1661

Reply:
It should. And it is empty. But if we know exactly the invocation script, why not to fill it in? Then we're getting closer to the real transaction witnesses.